internal/generate/packagemanifest/packagemanifest_test.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packagemanifest
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/format"
+
+	apimanifests "github.com/operator-framework/api/pkg/manifests"
+	"sigs.k8s.io/yaml"
+
+	genutil "github.com/operator-framework/operator-sdk/internal/generate/internal"
+)
+
+var (
+	testDataDir = filepath.Join("..", "testdata")
+)
+
+func TestGenerator(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Generator Suite")
+}
+
+var pkgManDefault, pkgManSingleChannel *apimanifests.PackageManifest
+
+var _ = BeforeSuite(func() {
+	initTestPackageManifestsHelper()
+})
+
+var _ = Describe("Generating a PackageManifest", func() {
+	format.UseStringerRepresentation = true
+
+	var (
+		g            Generator
+		buf          *bytes.Buffer
+		operatorName = "memcached-operator"
+		version      = "0.0.1"
+	)
+
+	BeforeEach(func() {
+		buf = &bytes.Buffer{}
+	})
+
+	Describe("for the new Go project layout", func() {
+
+		Context("with correct Options", func() {
+
+			var (
+				tmp string
+				err error
+			)
+
+			BeforeEach(func() {
+				tmp, err = ioutil.TempDir(".", "")
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				if tmp != "" {
+					os.RemoveAll(tmp)
+				}
+			})
+
+			It("should write a PackageManifest to an io.Writer", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					Version:      version,
+				}
+				opts := []Option{
+					WithBase(testDataDir),
+					WithWriter(buf),
+				}
+				Expect(g.Generate(opts...)).ToNot(HaveOccurred())
+				Expect(buf.String()).To(MatchYAML(pkgManDefaultContent))
+			})
+			It("should write a PackageManifest to a file", func() {
```

Comment:
think this should be "should write a PackageManifest to a filename" or something like that. I thought it meant an os.File at first, and was kinda unsure what the difference was from the previous test.